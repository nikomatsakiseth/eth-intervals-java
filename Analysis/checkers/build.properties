## This is a configuration file for use by Ant when building the JSR 308
## checkers and framework.

# The location of the JSR 308 OpenJDK distribution.
jsr308.langtools=../../jsr308-langtools
compiler.lib=${jsr308.langtools}/dist/lib/javac.jar:../jsr308-langtools/dist/lib/javac.jar
javadoc.lib=${jsr308.langtools}/dist/lib/javadoc.jar:${jsr308.langtools}/dist/lib/doclets.jar

# The location of the annotation file utilities.
annotation-utils.lib=../annotation-file-utilities/annotation-file-utilities.jar:lib/asmx.jar

javaparser.lib=lib/javaparser.jar

build.version = 1.0.4

jtreg.home= ${user.home}/research/jtreg

