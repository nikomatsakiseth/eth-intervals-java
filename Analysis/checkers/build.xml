<!--
  This is an Ant build file for compiling the Checker Framework.
-->
<project name="checkers" default="dist" basedir=".">

    <description>
      Builds the Checker Framework for the JSR 308 compiler.
    </description>

    <property name="src" value="src"/>
    <property name="tests" value="tests"/>
    <property name="tests.junit" value="${tests}/junit-4.7.jar"/>
    <property name="build" value="build"/>
    <property name="build.reports" value="${build}/reports"/>
    <property name="build.tests" value="tests/build"/>
    <property file="build.${os.name}.properties"/>
    <property file="build.properties"/>
    <property name="checkers.jar.file" value="checkers.jar"/>
    <property name="dist.annos.file" value="checkers-quals.jar"/>
    <property name="dist.doc" value="doc"/>
    <property name="javari.jdk.src" value="jdk/javari/src"/>
    <property name="javari.jdk.build" value="jdk/javari/build"/>
    <property name="javaparser.loc" value="../javaparser"/>
    <property name="checkers.rel.jp" value="../checkers"/>
    <property name="build.jtreg.dir" value="${build}/jtreg"/>

    <property name="tmpdir" value="${java.io.tmpdir}/${user.name}/${ant.project.name}" />

    <property name="compiler.version.goal" value="javac 1.7.0-jsr308-${build.version}"/>
    <property name="run.tests.should.fork" value="true"/>
    <property name="halt.on.test.failure" value="true"/>

    <path id="build.path">
        <pathelement location="${compiler.lib}"/>
    </path>

    <!-- Used to have:
           depends="touch-jdk"
         When is that truly necessary? -->
    <target name="prep" description="Create required directories">
        <mkdir dir="${build}"/>
        <mkdir dir="${build.tests}"/>
        <mkdir dir="${build.tests}/testclasses"/>
        <mkdir dir="${dist.doc}"/>

        <java fork="true"
              outputproperty="compiler.version"
              classpath="${compiler.lib}"
              classname="com.sun.tools.javac.Main">
          <arg line="-version"/>
        </java>

        <property name="compiler.exists" value="true"/>
        <condition property="compiler.exists">
           <equals
             arg1="${compiler.version}"
             arg2="${compiler.version.goal}"/>
        </condition>

        <available property="jp.project"
                   file="${javaparser.loc}/build.xml"/>
        <antcall target="-prep-jp-compile"/>
    </target>

    <target name="-prep-jp-compile" if="jp.project"
            description="compile javaparser project if available">
      <ant dir="${javaparser.loc}">
        <property name="dist.file"
                  value="${checkers.rel.jp}/${javaparser.lib}"/>
        <target name="dist"/>
      </ant>
    </target>

    <target name="clean" description="Remove generated files">
        <delete dir="${build}"/>
        <delete dir="${dist.doc}"/>
        <delete file="${checkers.jar.file}"/>
        <delete file="${dist.annos.file}"/>
        <delete dir="${temp.dir}"/>
        <delete file="${bindist.file}"/>
        <delete failonerror="false">
            <fileset dir="${build.tests}" includes="**/*.class"/>
            <fileset dir="jdk" includes="**/*.class"/>
        </delete>
    </target>

    <target name="build.check.uptodate"
            description="Set properties: filesets and build.uptodate">
    <fileset id="src.files" dir="${src}">
        <include name="**/*.java"/>
        <exclude name="checkers/util/dist/**"/>
    </fileset>
    <fileset id="src.astub.files" dir="${src}" includes="**/*.properties,**/*.astub"/>

    <!-- I can't figure out how to combine filesets (or even selectors)
             to get just one *.uptodate property, so do them individually
             and then combine with <condition>. -->
        <!-- Probably should depend on a file rather than a directory. -->
    <uptodate property="src.files.uptodate" targetfile="${build}">
      <srcfiles refid="src.files"/>
          <mapper type="glob" from="*.java" to="${build}/*.class"/>
    </uptodate>
    <uptodate property="src.astub.files.uptodate" targetfile="${build}">
      <srcfiles refid="src.astub.files"/>
    </uptodate>
    <uptodate property="javaparser.lib.uptodate" targetfile="${build}" srcfile="${javaparser.lib}"/>

    <condition property="build.uptodate">
      <and>
        <isset property="src.files.uptodate"/>
        <isset property="src.astub.files.uptodate"/>
        <isset property="javaparser.lib.uptodate"/>
      </and>
    </condition>
    <!--
    <echo message="src.files.uptodate: ${src.files.uptodate}"/>
    <echo message="src.astub.files.uptodate: ${src.astub.files.uptodate}"/>
    <echo message="javaparser.lib.uptodate: ${javaparser.lib.uptodate}"/>
    <echo message="build.uptodate: ${build.uptodate}"/>
    -->

    <fileset id="javari.jdk.src.files" dir=".">
        <include name="${javari.jdk.src}/**/*.java"/>
        </fileset>

    <!-- Probably will never be up to date, because of touch-jdk target. :-( -->
    <uptodate property="javari.jdk.uptodate">
      <srcfiles refid="javari.jdk.src.files"/>
          <mapper type="glob" from="${javari.jdk.src}/*.java" to="${javari.jdk.build}/*.class"/>
        </uptodate>
    <!--
    <echo message="javari.jdk.uptodate: ${javari.jdk.uptodate}"/>
    -->

    </target>

    <target name="build" depends="prep,build.check.uptodate" unless="build.uptodate"
            description="Compile files">
        <fail unless="compiler.exists"
            message="
Needed version ${compiler.version.goal} of the JSR 308 compiler,
but found version ${compiler.version} on your classpath.
${line.separator}${line.separator}
Check that the 'compiler.lib' property in 'build.properties' points to version
${build.version} of the 'javac.jar' library."/>

        <copy todir="${build}" preservelastmodified="true">
            <fileset refid="src.astub.files"/>
        </copy>
        <pathconvert pathsep=" " property="src.files.spaceseparated">
            <path>
                <fileset dir="${src}">
                    <include name="**/*.java"/>
                    <exclude name="checkers/util/dist/**"/>
                </fileset>
            </path>
        </pathconvert>
        <java fork="true"
              failonerror="true"
              classpath="${compiler.lib}:${annotation-utils.lib}:${javaparser.lib}"
              classname="com.sun.tools.javac.Main">
            <jvmarg line="-Xbootclasspath/p:${compiler.lib}"/>
            <arg value="-g"/>
            <arg line="-sourcepath ${src}"/>
            <arg line="-d ${build}"/>
            <arg line="${src.files.spaceseparated}"/>
            <arg line="-version"/>
        </java>
        <unjar src="${javaparser.lib}" dest="${build}" />
    <!--
    Touch doesn't work on a directory, so can't do:
       <touch file="${build}"/>
        Instead, create and remove a file, which modifies the directory.
    -->
    <touch file="${build}/.timestamp"/>
    <delete file="${build}/.timestamp"/>
    </target>

    <target name="build-javari-jdk" depends="build" unless="javari.jdk.uptodate"
    description="Compile Javari's annotated JDK.">
    <pathconvert pathsep=" " property="javari.jdk.src.files.spaceseparated">
       <path>
          <fileset refid="javari.jdk.src.files"/>
       </path>
    </pathconvert>
    <mkdir dir="${javari.jdk.build}"/>
    <java fork="true"
          failonerror="true"
          classpath="${build}:${compiler.lib}"
          classname="com.sun.tools.javac.Main">
       <jvmarg line="-Xbootclasspath/p:${compiler.lib}"/>
       <arg value="-g"/>
       <arg line="-sourcepath ${javari.jdk.src}"/>
       <arg line="-d ${javari.jdk.build}"/>
       <arg line="${javari.jdk.src.files.spaceseparated}"/>
       <arg line="-version"/>
    </java>
    </target>

    <target name="docs" depends="javadoc"/>

    <target name="javadoc" depends="prep" description="Create Javadoc documentation">
        <java fork="true"
              failonerror="true"
              classpath="${compiler.lib}:${javadoc.lib}:${java.home}/../lib/tools.jar"
              classname="com.sun.tools.javac.Main">
            <jvmarg line="-Xbootclasspath/p:${compiler.lib}:${javadoc.lib}"/>
            <arg value="-g"/>
            <arg line="-d ${build}"/>
            <arg line="src/checkers/util/dist/ManualTaglet.java"/>
            <arg line="-version"/>
        </java>
        <javadoc sourcepath="${src}" destdir="${dist.doc}" failonerror="true"
                 executable="../../jsr308-langtools/dist/bin/javadoc"
          classpath="${build}:${compiler.lib}:${javadoc.lib}"
          excludepackagenames="checkers.util.stub,checkers.util.dist"
          bootclasspath="${compiler.lib}:${javadoc.lib}:${java.home}/lib/rt.jar">
          <package name="checkers.*"/>
          <link href="http://download.java.net/jdk7/docs/api/"/>
          <link href="http://download.java.net/jdk7/docs/jdk/api/javac/tree/"/>
          <taglet name="checkers.util.dist.ManualTaglet" path="${build}"/>
        </javadoc>
    </target>

    <target name="dist" depends="jar,bindist">
    </target>

    <!-- This creates checkers.jar -->
    <target name="jar" depends="build,checkers-quals-jar"
            description="Create jar file">
        <jar destfile="${checkers.jar.file}" basedir="${build}"/>
    </target>

    <!-- This creates checkers-quals.jar -->
    <target name="checkers-quals-jar" depends="build"
            description="Create a jar file for annotations">
        <jar destfile="${dist.annos.file}" basedir="${build}">
            <include name="**/quals/*"/>
            <!-- Watch out for inner anonymous class within utilities -->
            <include name="**/NullnessUtils*"/>
        </jar>
    </target>

    <target name="tags" depends="prep" description="Create Emacs TAGS table">
        <exec executable="/bin/sh">
            <arg value="-c"/>
            <arg value="etags `find -name '*.java' | sort-directory-order.pl`"/>
        </exec>
    </target>

    <target name="build-tests" depends="prep" description="Compile tests">
        <fail unless="compiler.exists"
            message="
Version ${build.version} of the JSR 308 compiler was not found on your
classpath.
${line.separator}${line.separator}
Check that the 'compiler.lib' property in 'build.properties' points to version
${build.version} of the 'javac.jar' library."/>
        <pathconvert pathsep=" " property="src.tests">
            <path>
                <fileset dir="${tests}">
                    <include name="src/tests/**/*.java"/>
                </fileset>
            </path>
        </pathconvert>
        <java fork="true"
              failonerror="true"
              classpath="${compiler.lib}:${tests.junit}:${build}"
              classname="com.sun.tools.javac.Main">
            <jvmarg line="-Xbootclasspath/p:${compiler.lib}"/>
            <arg value="-g"/>
            <arg line="-sourcepath ${tests}"/>
            <arg line="-d ${build.tests}"/>
            <arg line="${src.tests}"/>
        </java>
    </target>

    <target name="-run-tests" description="Generalized test runner">
        <mkdir dir="${build.reports}"/>
        <junit fork="${run.tests.should.fork}"
            dir="${basedir}"
            printsummary="true"
            haltonfailure="${halt.on.test.failure}">
            <jvmarg line="-Xbootclasspath/p:${compiler.lib}"/>
            <jvmarg line="-ea"/>

            <classpath>
                <pathelement path="${compiler.lib}"/>
                <pathelement path="${checker.jar.file}"/>
                <pathelement path="${build.tests}"/>
                <pathelement path="${tests.junit}"/>
                <pathelement path="${build}"/>
            </classpath>
            <formatter type="xml"/>
            <formatter type="brief" usefile="false"/>

            <test name="${param}" todir="${build.reports}"/>
        </junit>
    </target>

    <!-- Note that this target ignores file src/tests/AllTests.java .
         We should perhaps eliminate either that file or this target.
     -->
    <target name="all-tests" depends="jar,build-tests"
      description="Run tests for all checkers">
      <!-- Copied from -run-tests target -->
      <mkdir dir="${build.reports}"/>
      <junit fork="${run.tests.should.fork}"
            dir="${basedir}"
          printsummary="true"
          haltonfailure="${halt.on.test.failure}">
          <jvmarg line="-Xbootclasspath/p:${compiler.lib}"/>
          <jvmarg line="-ea"/>

          <classpath>
              <pathelement path="${compiler.lib}"/>
              <pathelement path="${checker.jar.file}"/>
              <pathelement path="${build.tests}"/>
              <pathelement path="${tests.junit}"/>
              <pathelement path="${build}"/>
          </classpath>
          <formatter type="xml"/>
          <formatter type="brief" usefile="false"/>

          <batchtest todir="${build.reports}">
              <fileset dir="${tests}/src">
                  <include name="**/*Test.java"/>
                  <exclude name="**/AllTests.java"/>

                  <!-- Framework classes -->
                  <exclude name="**/CheckerTest.java"/>
                  <exclude name="**/ParameterizedCheckerTest.java"/>
              </fileset>
          </batchtest>
      </junit>
  </target>

    <target name="annotation-builder-tests" depends="jar,build-tests"
      description="Run tests for the Checker Framework">
        <antcall target="-run-tests">
            <param name="param" value="tests.AnnotationBuilderTest"/>
        </antcall>
    </target>

    <target name="basic-tests" depends="jar,build-tests"
      description="Run tests for the Checker Framework">
        <antcall target="-run-tests">
            <param name="param" value="tests.BasicEncryptedTest"/>
        </antcall>
        <antcall target="-run-tests">
            <param name="param" value="tests.BasicSuperSubTest"/>
        </antcall>
    </target>

    <target name="flow-tests" depends="jar,build-tests"
      description="Run tests for the Checker Framework">
        <antcall target="-run-tests">
            <param name="param" value="tests.FrameworkTest"/>
        </antcall>
    </target>

    <target name="framework-tests" depends="jar,build-tests"
      description="Run tests for the Checker Framework">
        <antcall target="-run-tests">
            <param name="param" value="tests.FrameworkTest"/>
        </antcall>
    </target>

    <target name="javari-tests" depends="jar,build-tests"
      description="Run tests for the Javari checker">
        <antcall target="-run-tests">
            <param name="param" value="tests.JavariTest"/>
        </antcall>
    </target>

    <target name="nullness-tests" depends="jar,build-tests"
      description="Run tests for the Nullness checker">
        <antcall target="-run-tests">
            <param name="param" value="tests.NullnessTest"/>
        </antcall>
    </target>

    <target name="interning-tests" depends="jar,build-tests"
      description="Run tests for the Interning checker">
        <antcall target="-run-tests">
            <param name="param" value="tests.InterningTest"/>
        </antcall>
    </target>

    <target name="igj-tests" depends="jar,build-tests"
      description="Run tests for the IGJ checker">
        <antcall target="-run-tests">
            <param name="param" value="tests.IGJTest"/>
        </antcall>
    </target>

    <target name="jtreg-tests" depends="-def-jtreg,jar">
        <jtreg-tool name="all" tests="."/>
    </target>

    <target name="-def-check">
        <macrodef name="check">
            <attribute name="name"/>
            <attribute name="property"/>
            <attribute name="marker"/>
            <sequential>
                <fail message="Cannot locate @{name}: please set @{property} to its location">
                    <condition>
                        <not>
                            <isset property="@{property}"/>
                        </not>
                    </condition>
                </fail>
                <fail message="@{name} is not installed in ${@{property}}">
                    <condition>
                        <not>
                            <available file="${@{property}}/@{marker}"/>
                        </not>
                    </condition>
                </fail>
            </sequential>
        </macrodef>
    </target>

    <target name="-check-jtreg.home" depends="-def-check">
        <check name="jtreg" property="jtreg.home" marker="lib/jtreg.jar"/>
    </target>

    <target name="-def-jtreg" unless="jtreg.defined" depends="-check-jtreg.home">
        <taskdef name="jtreg" classname="com.sun.javatest.regtest.Main$$Ant">
            <classpath>
                <pathelement location="${jtreg.home}/lib/jtreg.jar"/>
                <pathelement location="${jtreg.home}/lib/javatest.jar"/>
            </classpath>
        </taskdef>

        <macrodef name="jtreg-tool">
            <attribute name="name"/>
            <attribute name="tests"/>
            <!-- <attribute name="jdk" default="${java.home}"/> -->
            <attribute name="samevm" default="true"/>
            <attribute name="verbose" default="summary"/>
            <attribute name="options" default=""/>
            <attribute name="keywords" default="-keywords:!ignore"/>
            <attribute name="jpda.jvmargs" default=""/>
            <sequential>
                <property name="coverage.options" value=""/>    <!-- default -->
                <property name="coverage.classpath" value="${basedir}/${compiler.lib}:${basedir}/checkers.jar"/>    <!-- default -->
                <!-- removed:
                    jdk="@{jdk}"
                   -->
                <jtreg
                    dir="jtreg"
                    workDir="${build.jtreg.dir}/@{name}/work"
                    reportDir="${build.jtreg.dir}/@{name}/report"
                    samevm="@{samevm}" verbose="@{verbose}"
                    failonerror="false" resultproperty="jtreg.@{name}.result"
                    javacoptions="-g"
                    vmoptions="${coverage.options} -Xbootclasspath/p:${coverage.classpath}:${build.classes.dir} @{jpda.jvmargs}">
                    <arg line="@{keywords}"/>
                    <arg line="@{options}"/>
                    <arg line="@{tests}"/>
                </jtreg>
                <!-- the next two properties are for convenience, when only
                     a single instance of jtreg will be invoked. -->
                <condition property="jtreg.passed">
                    <equals arg1="${jtreg.@{name}.result}" arg2="0"/>
                </condition>
                <property name="jtreg.report" value="${build.jtreg.dir}/@{name}/report"/>
            </sequential>
        </macrodef>
        <property name="jtreg.defined" value="true"/>
    </target>

    <!-- Type Check the Checkers -->

    <!-- depends on jar, needs classfile of the checker itself -->
    <target name="-run-checker" depends="jar"
            description="Run Nullness checker on Checker Framework">
        <pathconvert pathsep=" " property="src.files">
            <path>
                <fileset dir="${src}">
                    <include name="**/*.java"/>
                    <exclude name="checkers/util/dist/**"/>
                </fileset>
            </path>
        </pathconvert>

        <java fork="true"
              failonerror="true"
              classpath="${compiler.lib}:${annotation-utils.lib}:${javaparser.lib}:${checkers.jar.file}"
              classname="com.sun.tools.javac.Main">
            <jvmarg line="-Xbootclasspath/p:${compiler.lib}"/>
            <arg value="-g"/>
            <arg line="-sourcepath ${src}:${checker-jdk}"/>
            <arg line="-d ${build}"/>
            <arg line="${src.files}"/>
            <arg line="-version"/>
            <arg line="-proc:only"/>
            <arg line="-processor ${checker-name}"/>
        </java>
    </target>

    <target name="check-nullness"
      description="Run the nullness checker on the Framework">
        <antcall target="-run-checker">
            <param name="checker-name" value="checkers.nullness.NullnessChecker"/>
            <param name="checker-jdk" value="jdk/nullness/src"/>
        </antcall>
    </target>

    <target name="check-igj"
      description="Run the IGJ checker on the Framework">
        <antcall target="-run-checker">
            <param name="checker-name" value="checkers.igj.IGJChecker"/>
            <param name="checker-jdk" value="jdk/igj/src"/>
        </antcall>
    </target>

    <!-- This throws off dependency tracking and causes rebuilding.
         When is it truly necessary? -->
    <!-- Why all JDK files?  Why not just source files? -->
    <target name="touch-jdk"
            description="Update the date of all jdk files">
        <touch>
            <fileset dir="jdk">
                <include name="**/*"/>
            </fileset>
        </touch>
    </target>

    <!--
      Binary release of the Checker Framework; it includes javac.
     -->

    <property name="javac.lib" value="${jsr308.langtools}/dist/lib/javac.jar"/>
    <property name="checkers.lib" value="checkers.jar"/>
    <property name="bindist.file" value="binary/jsr308-all.jar"/>
    <property name="temp.dir" value="binary/buildbin"/>

    <target name="-prep-bindist"
            description="ensures that the necessary jar files exist">
        <!-- check javac.jar -->
        <available property="javac.exist" file="${javac.lib}"/>
        <fail unless="javac.exist" message="Could not find javac.jar: ${javac.lib}"/>
    </target>

    <target name="bindist" depends="-prep-bindist,jar"
            description="Generates jar file jsr308-all.jar including both javac and the Checker Framework">
        <delete dir="${temp.dir}"/>
        <unjar src="${checkers.lib}" dest="${temp.dir}"/>
        <unjar src="${javac.lib}" dest="${temp.dir}"/>
        <jar destfile="${bindist.file}" basedir="${temp.dir}">
            <manifest>
                <attribute name="Main-Class" value="com.sun.tools.javac.Main"/>
            </manifest>
        </jar>
        <delete dir="${temp.dir}"/>
    </target>

</project>
