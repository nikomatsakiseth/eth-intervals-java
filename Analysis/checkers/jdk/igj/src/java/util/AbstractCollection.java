package java.util;

import checkers.igj.quals.*;

@I
public abstract class AbstractCollection<E> implements java.util.Collection<E> {
  protected AbstractCollection() @ReadOnly {}
  public abstract @I java.util.Iterator<E> iterator() @ReadOnly;
  public abstract int size() @ReadOnly;
  public boolean isEmpty() @ReadOnly { throw new RuntimeException("skeleton method"); }
  public boolean contains(java.lang.Object a1) @ReadOnly { throw new RuntimeException("skeleton method"); }
  public java.lang.Object[] toArray() @ReadOnly { throw new RuntimeException("skeleton method"); }
  public <T> T @Mutable [] toArray(T @Mutable [] a1) @ReadOnly { throw new RuntimeException("skeleton method"); }
  public boolean add(E a1) @Mutable { throw new RuntimeException("skeleton method"); }
  public boolean remove(java.lang.Object a1) @Mutable { throw new RuntimeException("skeleton method"); }
  public boolean containsAll(@ReadOnly java.util.Collection<?> a1) @ReadOnly { throw new RuntimeException("skeleton method"); }
  public boolean addAll(@ReadOnly java.util.Collection<? extends E> a1) @Mutable { throw new RuntimeException("skeleton method"); }
  public boolean removeAll(@ReadOnly java.util.Collection<?> a1) @Mutable { throw new RuntimeException("skeleton method"); }
  public boolean retainAll(@ReadOnly java.util.Collection<?> a1) @Mutable { throw new RuntimeException("skeleton method"); }
  public void clear() @Mutable { throw new RuntimeException("skeleton method"); }
  public java.lang.String toString() @ReadOnly { throw new RuntimeException("skeleton method"); }
}
