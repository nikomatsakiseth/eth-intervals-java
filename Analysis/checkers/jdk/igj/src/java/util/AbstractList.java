package java.util;

import checkers.igj.quals.*;

@I
public abstract class AbstractList<E> extends @I java.util.AbstractCollection<E> implements @I java.util.List<E> {
  protected AbstractList() @ReadOnly {}
  public boolean add(E a1) @Mutable { throw new RuntimeException("skeleton method"); }
  public abstract E get(int a1) @ReadOnly;
  public E set(int a1, E a2) @Mutable { throw new RuntimeException("skeleton method"); }
  public void add(int a1, E a2) @Mutable { throw new RuntimeException("skeleton method"); }
  public E remove(int a1) @Mutable { throw new RuntimeException("skeleton method"); }
  public int indexOf(@ReadOnly java.lang.Object a1) @ReadOnly { throw new RuntimeException("skeleton method"); }
  public int lastIndexOf(@ReadOnly java.lang.Object a1) @ReadOnly { throw new RuntimeException("skeleton method"); }
  public void clear() @Mutable { throw new RuntimeException("skeleton method"); }
  public boolean addAll(int a1, @ReadOnly java.util.Collection<? extends E> a2) @Mutable { throw new RuntimeException("skeleton method"); }
  public @I java.util.Iterator<E> iterator() @ReadOnly { throw new RuntimeException("skeleton method"); }
  public @I java.util.ListIterator<E> listIterator() @ReadOnly { throw new RuntimeException("skeleton method"); }
  public @I java.util.ListIterator<E> listIterator(int a1) @ReadOnly { throw new RuntimeException("skeleton method"); }
  public @I java.util.List<E> subList(int a1, int a2) @ReadOnly { throw new RuntimeException("skeleton method"); }
  public boolean equals(@ReadOnly java.lang.Object a1) @ReadOnly { throw new RuntimeException("skeleton method"); }
  public int hashCode() @ReadOnly { throw new RuntimeException("skeleton method"); }
}
