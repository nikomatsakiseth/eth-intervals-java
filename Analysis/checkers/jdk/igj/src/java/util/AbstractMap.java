package java.util;

import checkers.igj.quals.*;

@I
public abstract class AbstractMap<K, V> implements @I java.util.Map<K, V> {
  @I
  public static class SimpleEntry<K, V> implements @I java.util.Map.Entry<K, V>, @I java.io.Serializable {
      private static final long serialVersionUID = 0L;
    public SimpleEntry(K a1, V a2) @AssignsFields { throw new RuntimeException("skeleton method"); }
    public SimpleEntry(@I java.util.Map.Entry<? extends K, ? extends V> a1) @AssignsFields { throw new RuntimeException("skeleton method"); }
    public K getKey() @ReadOnly { throw new RuntimeException("skeleton method"); }
    public V getValue() @ReadOnly { throw new RuntimeException("skeleton method"); }
    public V setValue(V a1) @AssignsFields { throw new RuntimeException("skeleton method"); }
    public boolean equals(@ReadOnly java.lang.Object a1) @ReadOnly { throw new RuntimeException("skeleton method"); }
    public int hashCode() @ReadOnly { throw new RuntimeException("skeleton method"); }
    public java.lang.String toString() @ReadOnly { throw new RuntimeException("skeleton method"); }
  }

  @Immutable
  public static class SimpleImmutableEntry<K, V> implements @Immutable java.util.Map.Entry<K, V>, @Immutable java.io.Serializable {
      private static final long serialVersionUID = 0L;
    public SimpleImmutableEntry(K a1, V a2) @AssignsFields { throw new RuntimeException("skeleton method"); }
    public SimpleImmutableEntry(java.util.Map.Entry<? extends K, ? extends V> a1) @AssignsFields { throw new RuntimeException("skeleton method"); }
    public K getKey() @ReadOnly { throw new RuntimeException("skeleton method"); }
    public V getValue() @ReadOnly { throw new RuntimeException("skeleton method"); }
    public V setValue(V a1) @AssignsFields { throw new RuntimeException("skeleton method"); }
    public boolean equals(@ReadOnly java.lang.Object a1) @ReadOnly { throw new RuntimeException("skeleton method"); }
    public int hashCode() @ReadOnly { throw new RuntimeException("skeleton method"); }
    public java.lang.String toString() @ReadOnly { throw new RuntimeException("skeleton method"); }
  }

  protected AbstractMap() @ReadOnly {}
  public int size() @ReadOnly { throw new RuntimeException("skeleton method"); }
  public boolean isEmpty() @ReadOnly { throw new RuntimeException("skeleton method"); }
  public boolean containsValue(@ReadOnly java.lang.Object a1) @ReadOnly { throw new RuntimeException("skeleton method"); }
  public boolean containsKey(@ReadOnly java.lang.Object a1) @ReadOnly { throw new RuntimeException("skeleton method"); }
  public V get(java.lang.Object a1) @ReadOnly { throw new RuntimeException("skeleton method"); }
  public V put(K a1, V a2) @Mutable { throw new RuntimeException("skeleton method"); }
  public V remove(java.lang.Object a1) @Mutable { throw new RuntimeException("skeleton method"); }
  public void putAll(@ReadOnly java.util.Map<? extends K, ? extends V> a1) @Mutable { throw new RuntimeException("skeleton method"); }
  public void clear() @Mutable { throw new RuntimeException("skeleton method"); }
  public @I java.util.Set<K> keySet() @ReadOnly { throw new RuntimeException("skeleton method"); }
  public @I java.util.Collection<V> values() @ReadOnly { throw new RuntimeException("skeleton method"); }
  public abstract @I java.util.Set<@I java.util.Map.Entry<K, V>> entrySet() @ReadOnly;
  public boolean equals(@ReadOnly java.lang.Object a1) @ReadOnly { throw new RuntimeException("skeleton method"); }
  public int hashCode() @ReadOnly { throw new RuntimeException("skeleton method"); }
  public java.lang.String toString() @ReadOnly { throw new RuntimeException("skeleton method"); }
}
