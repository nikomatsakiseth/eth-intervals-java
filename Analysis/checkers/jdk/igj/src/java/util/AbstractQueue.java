package java.util;
import checkers.igj.quals.*;

@I
public abstract class AbstractQueue<E> extends @I java.util.AbstractCollection<E> implements @I java.util.Queue<E> {
  protected AbstractQueue() @ReadOnly {}
  public boolean add(E a1) @Mutable { throw new RuntimeException("skeleton method"); }
  public E remove() @Mutable { throw new RuntimeException("skeleton method"); }
  public E element() @ReadOnly { throw new RuntimeException("skeleton method"); }
  public void clear() @Mutable { throw new RuntimeException("skeleton method"); }
  public boolean addAll(@ReadOnly java.util.Collection<? extends E> a1) @Mutable { throw new RuntimeException("skeleton method"); }
}
