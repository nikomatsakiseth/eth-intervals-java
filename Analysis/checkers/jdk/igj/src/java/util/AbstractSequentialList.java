package java.util;
import checkers.igj.quals.*;

@I
public abstract class AbstractSequentialList<E> extends @I java.util.AbstractList<E> {
  protected AbstractSequentialList() @ReadOnly {}
  public E get(int a1) @ReadOnly { throw new RuntimeException("skeleton method"); }
  public E set(int a1, E a2) @Mutable { throw new RuntimeException("skeleton method"); }
  public void add(int a1, E a2) @Mutable { throw new RuntimeException("skeleton method"); }
  public E remove(int a1) @Mutable { throw new RuntimeException("skeleton method"); }
  public boolean addAll(int a1, @ReadOnly java.util.Collection<? extends E> a2) @Mutable { throw new RuntimeException("skeleton method"); }
  public @I java.util.Iterator<E> iterator() @ReadOnly { throw new RuntimeException("skeleton method"); }
  public abstract @I java.util.ListIterator<E> listIterator(int a1) @ReadOnly;
}
