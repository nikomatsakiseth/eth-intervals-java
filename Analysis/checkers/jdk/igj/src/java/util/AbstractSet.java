package java.util;
import checkers.igj.quals.*;

@I
public abstract class AbstractSet<E> extends @I java.util.AbstractCollection<E> implements @I java.util.Set<E> {
  protected AbstractSet() @ReadOnly {}
  public boolean equals(@ReadOnly java.lang.Object a1) @ReadOnly { throw new RuntimeException("skeleton method"); }
  public int hashCode() @ReadOnly { throw new RuntimeException("skeleton method"); }
  public boolean removeAll(@ReadOnly java.util.Collection<?> a1) @Mutable { throw new RuntimeException("skeleton method"); }
}
