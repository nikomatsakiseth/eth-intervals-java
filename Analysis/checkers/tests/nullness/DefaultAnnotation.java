import checkers.nullness.quals.*;
import checkers.quals.*;
import java.util.*;
public class DefaultAnnotation {

    public void testNoDefault() {

        String s = null;

    }

    @DefaultQualifiers(@DefaultQualifier(value="checkers.nullness.quals.NonNull", locations={DefaultLocation.ALL}))
    public void testDefault() {

        String s = null;                                // error
        List<String> lst = new List<String>();    // valid
        lst.add(null);                                  // error
    }

    @DefaultQualifier(value="checkers.nullness.quals.NonNull", locations={DefaultLocation.ALL})
    public class InnerDefault {

        public void testDefault() {
            String s = null;                                // error
            List<String> lst = new List<String>();    // valid
            lst.add(null);                                  // error
            s = lst.get(0);                                 // valid

            List<@Nullable String> nullList
                = new List<@Nullable String>();       // valid
            nullList.add(null);                             // valid
            s = nullList.get(0);                            // error
        }
    }

    @DefaultQualifier(value="checkers.nullness.quals.NonNull", locations={DefaultLocation.ALL})
    public static class DefaultDefs {

        public String getNNString() {
            return "foo";                               // valid
        }

        public String getNNString2() {
            return null;                                // error
        }

        public <T extends @Nullable Object> T getNull(T t) {
            return null;                                // invalid
        }

        public <T extends @NonNull Object> T getNonNull(T t) {
            return null;                                // error
        }
    }

    public class DefaultUses {

        public void test() {

            DefaultDefs d = new DefaultDefs();

            @NonNull String s = d.getNNString();        // valid

        }

        @DefaultQualifier(value="checkers.nullness.quals.NonNull", locations={DefaultLocation.ALL})
        public void testDefaultArgs() {

            DefaultDefs d = new DefaultDefs();

            String s1 = d.<@Nullable String>getNull(null);      // error
            String s2 = d.<String>getNonNull("foo");            // valid
            String s3 = d.<@Nullable String>getNonNull("foo");  // error
        }

    }

    @DefaultQualifier(value="checkers.nullness.quals.NonNull")
    static class DefaultExtends implements Iterator<String>, Iterable<String> {

        @Override public boolean hasNext() { throw new UnsupportedOperationException(); }
        @Override public void remove() { throw new UnsupportedOperationException(); }
        @Override public String next() { throw new UnsupportedOperationException(); }

        @Override
        public Iterator<String> iterator() {
            return this;
        }
    }

    class List<E extends @Nullable Object> {
        public E get(int i) { throw new RuntimeException(); }
        public boolean add(E e) { throw new RuntimeException(); }
    }

    @DefaultQualifier(value="NonNull")
    public void testDefaultUnqualified() {

        String s = null;                                // error
        List<String> lst = new List<String>();    // valid
        lst.add(null);                                  // error
    }

}
