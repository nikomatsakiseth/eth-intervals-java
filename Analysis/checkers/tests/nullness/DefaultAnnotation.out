:15: (type.incompatible)
:17: (type.incompatible)
:24: (type.incompatible)
:26: (type.incompatible)
:32: (type.incompatible)
:44: (type.incompatible)
:48: (type.incompatible)
:52: (type.incompatible)
:71: (type.incompatible)
:73: (type.incompatible)
:73: (generic.argument.invalid)
:99: (type.incompatible)
:101: (type.incompatible)
